### PR TITLE
Pin pytest to <5.4 in tox script

### PR DESCRIPTION
**Pull request by astrofrog**
This pins pytest to <5.4 in the tox script. Currently testing this in https://github.com/astropy/astropy/pull/10045

### CODE REVIEW

**Comment by pllim** [comment]
Seems to have fixed core Travis CI but not CircleCI.

**Comment by astrofrog** [comment]
@astrofrog that's a problem to solve in the astropy core PR. Can we merge this? :)

File: tox/run_tox.sh [hunk 1]
```diff
@@ -24,7 +26,8 @@ echo 'TOXENV='$TOXENV
 echo 'TOXARGS='$TOXARGS
 echo 'TOXPOSARGS='$TOXPOSARGS
 echo ''
-echo 'No global patches being applied'
+# echo 'No global patches being applied'
+echo 'Patching pytest to <5.4'
```
Comment (bsipocz) [inline tox/run_tox.sh]: add a comment here about the reason

**Comment by bsipocz** [comment]
Merging as last commit only added a comment.

thanks @astrofrog!

**Pull request merged**